### PR TITLE
SC-67

### PR DESCRIPTION
**Pull request by prestoncraw**
Added UI for configurable colors

### CODE REVIEW

File: Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MagDurCurves/CurveForm.tsx [hunk 1]
```diff
@@ -256,14 +269,14 @@ export default function CurveForm(props: IProps) {
                         zoomMode={'Rect'}
                         zoom={true} pan={false}
                         useMetricFactors={false} XAxisType={'log'}
-                        >
+                    >
                         <Line highlightHover={false} showPoints={false} lineStyle={'-'}
                             color={"#A30000"} data={curve} />
```
Comment (clackner-gpa) [inline Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MagDurCurves/CurveForm.tsx:274]:
```suggestion
                            color={props.Curve.Color} data={curve} />
```

File: Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MagDurCurves/CurveForm.tsx [hunk 2]
```diff
@@ -22,19 +22,17 @@
...
 import { ReactTable } from '@gpa-gemstone/react-table';
 import { Circle, Line, Plot } from '@gpa-gemstone/react-graph';
 import { DownArrow, TrashCan, UpArrow } from '@gpa-gemstone/gpa-symbols';
+import { ColorPicker } from '@gpa-gemstone/react-forms'
+import { OpenXDA } from '@gpa-gemstone/application-typings'
 
-declare var homePath: string;
-
-interface IProps { Curve: LocalXDA.IMagDurCurve, stateSetter: (tab: LocalXDA.IMagDurCurve) => void, setErrors?: (e: string[]) => void }
+interface IProps { Curve: OpenXDA.Types.MagDurCurve, stateSetter: (tab: LocalXDA.IMagDurCurve) => void, setErrors?: (e: string[]) => void }
```
Comment (clackner-gpa) [inline Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MagDurCurves/CurveForm.tsx:35]:
```suggestion
interface IProps { Curve: OpenXDA.Types.MagDurCurve, stateSetter: (curve: LocalXDA.IMagDurCurve) => void, setErrors?: (e: string[]) => void }
```

File: Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MagDurCurves/CurveForm.tsx [hunk 3]
```diff
@@ -256,14 +269,14 @@ export default function CurveForm(props: IProps) {
                         zoomMode={'Rect'}
                         zoom={true} pan={false}
                         useMetricFactors={false} XAxisType={'log'}
-                        >
+                    >
                         <Line highlightHover={false} showPoints={false} lineStyle={'-'}
                             color={"#A30000"} data={curve} />
-                        {curve.map((p,i) => (i == curve.length-1? null : <Circle data={[p[0], p[1]]} color={"#FFFFFF"} radius={8} borderColor={"#A30000"} borderThickness={2} text={String(curve.indexOf(p) + 1)} key={i} />))}
-                    </Plot> 
+                        {curve.map((p, i) => (i == curve.length - 1 ? null : <Circle data={[p[0], p[1]]} color={"#FFFFFF"} radius={8} borderColor={"#A30000"} borderThickness={2} text={String(curve.indexOf(p) + 1)} key={i} />))}
```
Comment (clackner-gpa) [inline Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MagDurCurves/CurveForm.tsx:275]:
```suggestion
                        {curve.map((p, i) => (i == curve.length - 1 ? null : <Circle data={[p[0], p[1]]} color={props.Curve.Color} radius={8} borderColor={props.Curve.Color} borderThickness={2} text={String(curve.indexOf(p) + 1)} key={i} />))}
```

File: Source/Applications/SystemCenter/package.json [hunk 4]
```diff
@@ -46,6 +46,7 @@
     "prop-types": "15.7.2",
     "query-string": "5.1.1",
     "react": "18.2.0",
+    "react-color": "^2.19.3",
```
[the remark this answers is not in the document]
Comment (prestoncraw) [reply Source/Applications/SystemCenter/package.json:49]: Yep, I missed that

**Pull request merged**